lib/puppet.rb

```diff
@@ -36,6 +37,10 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    include FastGettext::Translation
+    GettextSetup.initialize(File.absolute_path('../locales', File.dirname(__FILE__)))
+    FastGettext.locale = GettextSetup.negotiate_locale(ENV["HTTP_ACCEPT_LANGUAGE"])
```

Comment:
Good question. Let me verify what might be the right way to determine
preferred locale from a command-line app.

On Mon, Jul 25, 2016 at 3:46 PM, Adrien Thebo <notifications@github.com>
wrote:

> In lib/puppet.rb
> <https://github.com/puppetlabs/puppet/pull/5136#discussion_r72159001>:
>
> > @@ -36,6 +37,10 @@ module Puppet
> >    require 'puppet/environments'
> >
> >    class << self
> > +    include FastGettext::Translation
> > +    GettextSetup.initialize(File.absolute_path('../locales', File.dirname(__FILE__)))
> > +    FastGettext.locale = GettextSetup.negotiate_locale(ENV["HTTP_ACCEPT_LANGUAGE"])
>
> Under what conditions do we expect ENV["HTTP_ACCEPT_LANGUAGE"] to be set?
> I would expect that this would be set when servicing an HTTP request, but
> this seems funny to always set even outside of a web service request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/5136/files/0e3b44a0fa02570f26affb9c006a9b0072b796e4#r72159001>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMVyKoXikajf5nCnDy27DlpDCU6HCnPUks5qZTzOgaJpZM4JTMHZ>
> .
>



-- 
*Andy Henroid*
Senior Software Engineer
andy.henroid@puppet.com